sanic/headers.py

```diff
@@ -33,175 +34,310 @@
 # For more information, consult ../tests/test_requests.py
 
 
-def parse_arg_as_accept(f):
-    def func(self, other, *args, **kwargs):
-        if not isinstance(other, Accept) and other:
-            other = Accept.parse(other)
-        return f(self, other, *args, **kwargs)
+class MediaTypePart(str):
+    @property
+    def is_wildcard(self) -> bool:
+        return str.__eq__(self, "*")
+
+    def eq(self, other: Any) -> bool:
+        return (
+            self == other
+            or self.is_wildcard
+            or MediaTypePart(other).is_wildcard
+        )
 
-    return func
 
+class MediaType:
+    """A media type, as used in the Accept header."""
 
-class MediaType(str):
-    def __new__(cls, value: str):
-        return str.__new__(cls, value)
+    def __init__(
+        self,
+        type_: str,
+        subtype: str,
+        **params: str,
+    ):
+        self.type_ = MediaTypePart(type_)
```

Comment:
Property name doesn't need underscore

Reply:
Pretty sure that is a copy/paste change.